### ISSUE TITLE
Faces inconsistency

### ISSUE THREAD

**Issue by manuel-uberti** [emacs-helm/helm-ls-git#55]
Hi,

I am using latest `helm` and `helm-ls-git` (I cloned the repositories with `straight`, so no `package.el` involved).

Starting from `emacs -Q`, I installed `helm` and `helm-ls-git` (again, using `straight`). `M-x helm-browse-project` gives me this:

![1](https://user-images.githubusercontent.com/8199224/70846811-80357400-1e5d-11ea-8f2e-90ea2ed36056.png)

While `helm-M-x` gives me this:

![2](https://user-images.githubusercontent.com/8199224/70846818-8af00900-1e5d-11ea-8d0b-6e783082d43a.png)

I was expecting to see `Buffers in git project (master)` and `Git files (master)` with a similar (the same?) face as `Emacs Commands History` and `Emacs Commands`.

This is the Emacs version I am using:
```
GNU Emacs 27.0.50 (build 1, x86_64-debian-linux-gnu, GTK+ Version 3.22.30)
 of 2019-12-14

Repository revision: c6e655c77ba1cee15ce3b7d8ed60069554fe3d45
Configured using:
--host=x86_64-debian-linux-gnu --with-modules --with-imagemagick --with-harfbuzz
```

If you need me to provide more details and/or do further tests, feel free to ask.

**Comment by manuel-uberti** [comment]
Actually, the same is happening for `helm-find-files` too.

![1](https://user-images.githubusercontent.com/8199224/70848334-46219d80-1e70-11ea-8f63-270e2fc3d99f.png)

Note that this is happening whether `helm-mode` is turned on or not.

**Comment by thierryvolpiatto** [comment]
This is related to https://github.com/emacs-helm/helm/issues/2213 and there is a long standing thread on emacs-dev about this (one of the issues that prevent emacs-27 starting pretest).
This related to the :extend face attribute.

**Comment by thierryvolpiatto** [comment]
That annoy everybody...

**Comment by manuel-uberti** [comment]
I see, thanks for the tip. I tried adding `:extend t` to `helm-source-header` in the theme I use but nothing changes. I'll try apply the function from the issue you linked.

**Comment by manuel-uberti** [comment]
That function doesn't seems to fix the problem though.

**Comment by thierryvolpiatto** [comment]
Manuel Uberti <notifications@github.com> writes:

> That function doesn't seems to fix the problem though.

Hmm, perhaps :extend code have changed in emacs-27, I will try to update
ASAP to try. I am actually on 7e387c9e526.
Perhaps your theme is affected as well (most themes are actually).

-- 
Thierry

Get my Gnupg key:
gpg --keyserver pgp.mit.edu --recv-keys 59F29997

**Comment by thierryvolpiatto** [comment]
Indeed, I installed a fresh emacs-27 and many faces are broken now with the last changes in the `:extend` code.

**Comment by manuel-uberti** [comment]
Thank you for taking the time to look into this. I opened a bug [here](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=38633) referring to this ticket.

**Comment by thierryvolpiatto** [comment]
See https://github.com/emacs-helm/helm/issues/2213 for more details.

**Comment by manuel-uberti** [comment]
Thanks to your recipe, I think Eli found the problem: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=38633#44

**Comment by manuel-uberti** [comment]
This now appears to be fixed by commit `1e240a0ecd` on Emacs master. 

I'm closing this issue, then. Thanks again for helping me debug it.

**Comment by thierryvolpiatto** [comment]
I confirm it is now working fine, thanks.
